pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -64,13 +62,85 @@ type Combinations []Combination
 
 // FanOut returns an list of params that represent combinations
 func (m *Matrix) FanOut() []Params {
-	var combinations Combinations
+	var combinations, includeCombinations Combinations
+	includeCombinations = m.getIncludeCombinations()
+	if m.HasInclude() && !m.HasParams() {
+		// If there are only Matrix Include Parameters return explicit combinations
+		return includeCombinations.toParams()
+	}
+	// Generate combinations from Matrix Parameters
 	for _, parameter := range m.Params {
 		combinations = combinations.fanOutMatrixParams(parameter)
 	}
+	combinations = combinations.overwriteCombinations(includeCombinations)
```

Comment:
I do not think we need to assign the returned `Combinations` to `combinations`, my understanding is, the `overwriteCombinations` is updating the `combinations`, can you please confirm? thanks!

Reply:
I looked into it further, the member receiver does not propagate the changes unless its a pointer receiver:

Changing:

```
func (cs Combinations) overwriteCombinations(ics Combinations) Combinations {
```

to:

```
func (cs *Combinations) overwriteCombinations(ics Combinations) {
```

Not blocking merging this PR for the suggested changes ⬆️ 

thanks!

Reply:
@pritidesai  Yes you are correct. A PR has been opened separately here: https://github.com/tektoncd/pipeline/pull/6463